client/requestUtil.js

```diff
@@ -420,6 +429,49 @@ RequestUtil.prototype.put = function (category, record) {
   })
 }
 
+/**
+ * Compact all records in a category
+ * @param {string=} category - the category ID
+ */
+RequestUtil.prototype.compactCategory = function (category) {
+  const thisCategory = category
+  if (!recordUtil.CATEGORY_IDS.includes(thisCategory)) {
+    throw new Error(`Unsupported sync category: ${category}`)
+  }
+
+  const prefix = `${this.apiVersion}/${this.userId}/${category}`
+  let options = {
+    MaxKeys: 1000,
```

Comment:
Do we list 1000 objects at a time? For example we have 10K records and a client sends that command. We will check only 1000 records. There could be a situation when we have 2K unique records and 8K need to be compacted, but we pull 1000 unique always and this will never be compacted. The list command returns `nextContinuationToken` which should be used on a next execution to avoid getting same records.

Reply:
https://github.com/brave/sync/pull/350/files#diff-93951cfb12e03132f0bd5b9a600a0a27R449
I thought specify `limitResponse` to false and it will call the `listObjectsV2` recursively with `nextContinuationToken ` untill we don't see `isTruncated`

Reply:
so theoretically, we will get all records down for 1000 each batch, the listObjectsV2  API maximum key limit is also 1000

Reply:
ah yeah, missed the part that you pass `false` there